src/core/syscall_info.rs

```diff
@@ -0,0 +1,138 @@
+use crate::core::errors::syscall_hadler_errors::SyscallHandlerError;
+use crate::core::errors::syscall_hadler_errors::SyscallHandlerError::*;
+use cairo_rs::serde::deserialize_program::*;
+use num_bigint::BigInt;
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) enum SyscallType {
+    EmitEvent {
+        selector: Option<Member>,
+        keys_len: Option<Member>,
+        keys: Option<Member>,
+        data_len: Option<Member>,
+        data: Option<Member>,
+    },
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub(crate) struct SyscallInfo {
+    selector: BigInt,
+    syscall_size: u32,
+    syscall_struct: SyscallType,
+}
+
+fn get_selector(
+    selector: &str,
+    identifiers: &HashMap<String, Identifier>,
+) -> Result<BigInt, SyscallHandlerError> {
+    identifiers
+        .get(selector)
+        .ok_or(MissingMember)?
+        .to_owned()
+        .value
+        .ok_or(MissingSelector)
+}
+
+fn get_identifier(
+    syscall: &str,
+    identifiers: &HashMap<String, Identifier>,
+) -> Result<Identifier, SyscallHandlerError> {
+    Ok(identifiers
+        .get(syscall)
+        .ok_or(MissingIdentifiers)?
+        .to_owned())
+}
+
+fn get_member(key: &str, members: &HashMap<String, Member>) -> Option<Member> {
+    members.get(key).map(ToOwned::to_owned)
+}
+
+impl SyscallInfo {
+    pub fn emit_event(
+        identifiers: &HashMap<String, Identifier>,
+    ) -> Result<SyscallInfo, SyscallHandlerError> {
+        let identifier = get_identifier("__main__.EmitEvent", identifiers)?;
+        let selector = get_selector("__main__.EMIT_EVENT_SELECTOR", identifiers)?;
+        let members = identifier.members.ok_or(MissingMember)?;
+
+        let syscall_type_selector = get_member("selector", &members);
+        let keys_len = get_member("keys_len", &members);
+        let keys = get_member("keys", &members);
+        let data_len = get_member("data_len", &members);
+        let data = get_member("data", &members);
+
+        Ok(SyscallInfo {
+            selector,
+            syscall_size: 5,
+            syscall_struct: SyscallType::EmitEvent {
+                selector: syscall_type_selector,
+                keys_len,
+                keys,
+                data_len,
+                data,
+            },
+        })
+    }
+}
+
+pub fn program_json() -> Result<ProgramJson, SyscallHandlerError> {
+    let path = Path::new(file!())
+        .parent()
+        .ok_or(MissingSyscallsJsonFile)?
+        .join("../../cairo_syscalls/syscalls.json");
+
+    let file = fs::read_to_string(path).unwrap();
+    serde_json::from_str(&file).map_err(|_| MissingSyscallsJsonFile)
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
Since we currently allow for missing members and report errors in those cases, unless we change that we need a test of the failure.